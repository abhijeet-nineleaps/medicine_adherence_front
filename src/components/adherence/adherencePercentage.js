const AdherencePercentage = (startDate, days, times, currentCount) => {
  return new Promise(res => {
    let tilldatecount = 0;
    let msPerDay = 86400000;
    const weeks = ['Sun', 'Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat'];
    const today = new Date();
    let startingDate = new Date(startDate);
    const utc1 = Date.UTC(
      today.getFullYear(),
      today.getMonth(),
      today.getDate(),
    );
    const utc2 = Date.UTC(
      startingDate.getFullYear(),
      startingDate.getMonth(),
      startingDate.getDate(),
    );
    let left = (utc2 - utc1) / msPerDay;
    let daysSet = new Set(days.split(':'));
    while (left >= 0) {
      if (daysSet.has(weeks[startingDate.getDay()])) {
        tilldatecount += times.split('-').length;
      }
      left--;
      startingDate.setDate(startingDate.getDate() + 1);
    }
    res(Math.round((currentCount / tilldatecount) * 100));
  });
};

export default AdherencePercentage;
