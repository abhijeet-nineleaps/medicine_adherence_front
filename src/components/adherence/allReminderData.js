import globalDb from '../../repositories/database/globalDb';
import Logger from '../logger';

const db = globalDb();
const allreminderdata = async (med_name) => {
  let reminder_obj;
  let map = new Map();
  let med_id = 0;

  function reminder_promise() {
    return new Promise(resolve => {
      db.transaction(async function (txn) {
        txn.executeSql(
          'CREATE TABLE IF NOT EXISTS User_medicines(user_id INTEGER PRIMARY KEY NOT NULL, medicine_name TEXT, medicine_des TEXT , title TEXT, time TEXT , days TEXT , start_date TEXT , end_date TEXT , status INTEGER , sync INTEGER)',
          [],
        );
        txn.executeSql(
          'SELECT * FROM `User_medicines` WHERE medicine_name = ?',
          [med_name],
          function (_tx, res) {
            reminder_obj = res.rows.item(0);
            med_id = parseInt(res.rows.item(0).user_id);
            txn.executeSql(
              'SELECT * FROM `reminder_day` WHERE med_id = ?',
              [med_id],
              function (_txx, respp) {
                for (let o = 0; o < respp.rows.length; o++) {
                  const curr_rem_obj = respp.rows.item(o);

                  let overall_timings = new Set(reminder_obj.time.split('-'));
                  let taken_missed_times = new Set(
                    curr_rem_obj.timings.split('-'),
                  );
                  let final_timeings_obj = {
                    remId: respp.rows.item(o).rem_id,
                    taken: [],
                    not_taken: [],
                  };

                  overall_timings.forEach((m_time) => {
                    if (taken_missed_times.has(m_time)) {
                      final_timeings_obj.not_taken.push(m_time);
                    } else {
                      final_timeings_obj.taken.push(m_time);
                    }
                  });
                  map.set(curr_rem_obj.date, final_timeings_obj);
                }
                resolve(map);
              },
            );
          },
        );
      });
    });
  }
  map = await reminder_promise();

  Logger.loggerInfo(map);
  return {mapper: map, meds_id: med_id};
};

export default allreminderdata;
