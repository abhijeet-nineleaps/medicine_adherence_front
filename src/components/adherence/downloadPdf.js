import {API_URL} from '../../repositories/var';
import RNFetchBlob from 'rn-fetch-blob';
import {logger} from 'react-native-logs';
import Logger from '../logger';

const {config} = RNFetchBlob;

const DownloadPdf = async (globalmedId) => {
  const date = new Date();

  let downloaddir = RNFetchBlob.fs.dirs.DownloadDir;
  let downloadPath = `${downloaddir}/report_${Math.floor(
    date.getTime() + date.getSeconds() / 2,
  )}.pdf`;
  const options = {
    fileCache: true,
    addAndroidDownloads: {
      useDownloadManager: true, // true will use native manager and be shown on notification bar.
      notification: true,
      path: downloadPath,
      description: 'Downloading your report',
    },
  };
  let generate_pdf_url = `${API_URL}/api/v1/pdf?medId=${globalmedId}`;
  let response = '';

  try {
    await fetch(generate_pdf_url)
      .then(res => res.json())
      .then(async resp => {
        Logger.loggerInfo(resp.message);
        await config(options)
          .fetch('GET', `${API_URL}/upload/static/pdf/${resp.message}.pdf`)
          .catch(() => {
            return 'err';
          });
      })
      .catch(() => {
        response = 'err';
        return response;
      });
    return response;
  } catch (err) {
    response = 'err';
    return response;
  }
};

export default DownloadPdf;
