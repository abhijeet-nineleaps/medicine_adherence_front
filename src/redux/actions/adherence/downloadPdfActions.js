import Types from '../allTypes';

function downloadPdf(globalmedId) {
  return {
    type: Types.PDF_DOWNLOAD,
    payload: globalmedId,
  };
}

function downloadPdfSuccess(data) {
  return {
    type: Types.SUCCESS_PDF_DOWNLOAD,
    payload: data,
  };
}

function downloadPdfError(error) {
  return {
    type: Types.FAILED_PDF_DOWNLOAD,
    payload: error,
  };
}

export const downloadPdfActions = {
  downloadPdf,
  downloadPdfSuccess,
  downloadPdfError,
};
