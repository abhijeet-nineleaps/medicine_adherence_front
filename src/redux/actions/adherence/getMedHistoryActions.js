import Types from '../allTypes';

function getMedHistory(globalmedId) {
  return {
    type: Types.GET_MED_HISTORY,
    payload: globalmedId,
  };
}

function medHistorySuccess(data) {
  return {
    type: Types.GET_MED_HISTORY_SUCCESS,
    payload: data,
  };
}

function medHistoryError(error) {
  return {
    type: Types.GET_MED_HISTORY_FAILED,
    payload: error,
  };
}

export const getMedHistoryActions = {
  getMedHistory,
  medHistorySuccess,
  medHistoryError,
};
