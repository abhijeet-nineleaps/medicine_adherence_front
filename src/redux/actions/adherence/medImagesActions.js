import Logger from '../../../components/logger';
import Types from '../allTypes';


function fetchMedImages(user_id) {
  return {
    type: Types.GET_MED_IMAGES,
    payload: user_id,
  };
}
function fetchMedImagesSuccess(data) {
  Logger.loggerInfo(data);
  return {
    type: Types.SUCCESS_MED_IMAGES,
    payload: data,
  };
}
function fetchMedImagesError(error) {
  Logger.loggerError(error);
  return {
    type: Types.FAILED_MED_IMAGES,
    payload: error,
  };
}

export const medImagesActions = {
  fetchMedImages,
  fetchMedImagesSuccess,
  fetchMedImagesError,
};
