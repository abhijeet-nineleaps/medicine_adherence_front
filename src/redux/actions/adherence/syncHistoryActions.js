import Logger from '../../../components/logger';
import Types from '../allTypes';
function syncHistory(params) {
  return {
    type: Types.SYNC_HISTORY,
    payload: params,
  };
}
function syncHitsorySuccess(data) {
  Logger.loggerInfo(data);
  return {
    type: Types.SUCCESS_SYNC_HISTORY,
    payload: data,
  };
}
function syncHistoryError(error) {
  Logger.loggerError(error);
  return {
    type: Types.FAILED_SYNC_HISTORY,
    payload: error,
  };
}

export const syncHistoryActions = {
  syncHistory,
  syncHitsorySuccess,
  syncHistoryError,
};
