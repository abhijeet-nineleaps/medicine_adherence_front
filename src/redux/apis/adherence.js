import axios from './axios';
import { API_URLS } from '../../constants/apiUrl';

const adherence = {
  syncmedicineHistory: async (params) => {
    const {meds_id, syncData} = params;
    let response = await fetch(`${API_URLS.SYNC_HISTORY}?medId=${meds_id}`, {
      method: 'POST',
      body: JSON.stringify(syncData),
      headers: {
        'Content-type': 'application/json',
      },
    });
    return response;
  },
  medimages: async function fetchmedimages(payload) {
    const response = await axios.get(`${API_URLS.MED_IMAGES}?medId=${payload}`);
    return response.data;
  },
  downloadPdf: async function downloadPdf(payload) {
    const response = await axios.get(`${API_URLS.DOWNLOAD_PDF}?medId=${payload}`);
    return response.data;
  },
  syncmeds: async function syncMeds(payload) {
    const response = await axios.get(`${API_URLS.SYNC_MEDS}=${payload}`);
    return response.data;
  },
  getmedhistory: async function getMedHistory(payload) {
    const response = await axios.get(`${API_URLS.GET_MED_HISTORY}?medId=${payload}`);
    return response.data;
  }

};


export default adherence;
