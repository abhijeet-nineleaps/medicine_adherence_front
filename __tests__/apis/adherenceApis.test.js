import adherence from '../../src/redux/apis/adherence';
describe('test adherence apis', () => {
  it('test medimages', () => {
    const payload = 'payload';
    expect(adherence.medimages(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });
 

  it('test downloadpdf', () => {
    const payload = 'payload';
    expect(adherence.downloadPdf(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });

  it('test getmedhistory', () => {
    const payload = 'payload';
    expect(adherence.getmedhistory(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });

  it('test syncmeds', () => {
    const payload = 'payload';
    expect(adherence.syncmeds(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });
  it('test syncmedhistory', () => {
    const payload = 'payload';
    expect(adherence.syncmedicineHistory(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });
});
