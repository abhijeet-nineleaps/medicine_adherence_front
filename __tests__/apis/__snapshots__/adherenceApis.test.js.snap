// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test syncmedhistory test the only function 1`] = `undefined`;
